packages/dcos-integration-test/extra/conftest.py

```diff
@@ -30,7 +30,10 @@ def dcos_api_session(dcos_api_session_factory):
         exhibitor_admin_password=exhibitor_admin_password,
         **args)
     api.wait_for_dcos()
-    return api
+    try:
+        yield api
+    finally:
+        api.session.close()
```

Comment:
Should it check if `api.sessios != None`?

Reply:
We'd expect that there should always be a session object at this point, right? So if there isn't one then raising an error is what we want? (Errors are still raised in `finally` blocks.)

```
16:54:20 margaretsy ~/mesosphere/dcos:$ python3
Python 3.5.0 (v3.5.0:374f501f4567, Sep 12 2015, 11:00:19) 
[GCC 4.2.1 (Apple Inc. build 5666) (dot 3)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> try:
...     print('blah')
... finally:
...     1 + 'nope'
... 
blah
Traceback (most recent call last):
  File "<stdin>", line 4, in <module>
TypeError: unsupported operand type(s) for +: 'int' and 'str'
>>> 
```